pkgs/applications/misc/wmenu/default.nix

```diff
@@ -0,0 +1,43 @@
+{ lib
+, stdenv
+, fetchFromSourcehut
+, pkg-config
+, cmake
+, meson
+, ninja
+, cairo
+, pango
+, wayland
+, wayland-scanner
+, wayland-protocols
+, libxkbcommon
+, scdoc
+}:
+
+stdenv.mkDerivation rec {
+  pname = "wmenu";
+  version = "0.1.2";
+
+  src = fetchFromSourcehut {
+    owner = "~adnano";
+    repo = "wmenu";
+    rev = version;
+    hash = "sha256-mS4qgf2sjgswasZXsmnbIWlqVv+Murvx1/ob0G3xsws=";
+  };
+
```

Comment:
I tried applying patch from upstream master branch like so

```patch
+  patches = [
+    (fetchpatch {
+      url = "https://git.sr.ht/~adnano/wmenu/commit/ba10072cdec9b0d4b51bcf305ff27dcf3003ae42.patch";
+      hash = "sha256-XF7xmEnsKlExMJQ5iS7wQG9Ja6ocrR0YvQuWFfByKVA=";
+    })
+  ];
+

```

and it still crashes for me

```
./result/bin/wmenu 
wmenu: ../main.c:1020: menu_init: Assertion `state->layer_shell != NULL' failed.
Aborted (core dumped)

```